src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -170,6 +184,13 @@ public class SqlMutationEngine : IMutationEngine
                 throw _dabExceptionWithTransactionErrorMessage;
             }
 
+            if (!isReadPermissionConfigured)
+            {
+                throw new DataApiBuilderException(message: "The database operation was successful but the current user is unable to view the response due to lack of read permissions",
+                                                  statusCode: HttpStatusCode.Unauthorized,
```

Comment:
```suggestion
                                                  statusCode: HttpStatusCode.Forbidden,
```

Forbidden maps to HTTP 403 -> Authorization error
Unauthorized maps to HTTP 401 -> Unauthenticated/Authentication error.